satellite/audit/reservoirservice.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/metainfo"
+)
+
+// ReservoirService is a temp name for the service struct during the audit 2.0 refactor.
+// Once V3-2363 and V3-2364 are implemented, Service2 will replace the existing Service struct.
+type ReservoirService struct {
+	log *zap.Logger
+
+	nodesToSelect  int
+	PathsToAudit   []storj.Path
+	reservoirSlots int
+	Reservoirs     map[storj.NodeID]*Reservoir
+	rand           *rand.Rand
+
+	MetainfoLoop *metainfo.Loop
+	Loop         sync2.Cycle
+}
+
+// NewReservoirService instantiates Service2
+func NewReservoirService(log *zap.Logger, metaLoop *metainfo.Loop, config Config) (*ReservoirService, error) {
+	return &ReservoirService{
+		log: log,
+
+		nodesToSelect:  config.NodesToSelect,
+		reservoirSlots: config.Slots,
+		rand:           rand.New(rand.NewSource(time.Now().Unix())),
+
+		MetainfoLoop: metaLoop,
+		Loop:         *sync2.NewCycle(config.Interval),
+	}, nil
+}
+
+// Run runs auditing service 2.0
+func (service *ReservoirService) Run(ctx context.Context) (err error) {
```

Comment:
Rather than `Run` and `Select` I think you could call them `RunPopulateReservoirs` and `RunReservoirAudits` or something. I think `Select` does not accurately explain what is being done inside it.